Model/Api/OAuthRedirect.php

```diff
@@ -351,10 +369,17 @@ public function login($code = '', $scope = '', $state = '', $reference = '')
                 // The checkout may not have been completed yet, but the user may have logged in via Bolt SSO
                 $quote = $this->cartHelper->getQuoteById($reference);
                 if ($quote !== false) {
+                    try {
+                        // we should make current customer cart as inactive and use guest cart as customer cart if login from bolt modal
+                        $customerActiveQuote = $this->cartRepository->getActiveForCustomer($customer->getId());
+                        $customerActiveQuote->setIsActive(false);
+                        $this->cartHelper->saveQuote($customerActiveQuote);
+                        // phpcs:ignore Magento2.CodeAnalysis.EmptyBlock
+                    } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
```

Comment:
Can we at least log this exception it may help with debugging issues in the future? like a bugsnag log.

Reply:
Added bugsnag log exception.